gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegister.java

```diff
@@ -109,46 +88,29 @@ public HiveRegister(State state) throws IOException {
       @Override
       public Void call() throws Exception {
 
-        IMetaStoreClient client = borrowClient();
-
-        try {
-
-          if (spec instanceof HiveSpecWithPredicates && !evaluatePredicates((HiveSpecWithPredicates) spec)) {
-            log.info("Skipping " + spec + " since predicates return false");
-            return null;
-          }
+        if (spec instanceof HiveSpecWithPredicates && !evaluatePredicates((HiveSpecWithPredicates) spec)) {
+          log.info("Skipping " + spec + " since predicates return false");
+          return null;
+        }
 
-          if (spec instanceof HiveSpecWithPreActivities) {
-            for (Activity activity : ((HiveSpecWithPreActivities) spec).getActivities()) {
-              activity.execute(HiveRegister.this);
-            }
+        if (spec instanceof HiveSpecWithPreActivities) {
+          for (Activity activity : ((HiveSpecWithPreActivities) spec).getPreActivities()) {
+            activity.execute(HiveRegister.this);
           }
+        }
 
-          Table table = spec.getTable();
-          createDbIfNotExists(client, table.getDbName());
-
-          Optional<Partition> partition = spec.getPartition();
-          createOrAlterTable(client, table);
-
-          if (partition.isPresent()) {
-
-            // Register a partition
-            addOrAlterPartition(client, table, partition.get(), spec.getPath());
```

Comment:
Yes for `createDbIfNotExists`. For the other two, if the table/partition already exists, there are different ways to determine whether the table/partition should be updated. So they are a little ambiguous to be part of the HiveRegister API.